### DISCUSSION TITLE
New line escape sequance for multi line header title?

### DISCUSSION THREAD

**Discussion by MikeLemo1** [junegunn/fzf#3664] [Q&A] [answered]
what is the escape sequance insside the `--header ' '` string that creates a new line?
It's usually `\n` but doesn'y seem to work other than a literal new line in nvim which is inconvinient 
and can create syntax error.
I also added `--header-lines 2` if that does anything related I Just need 2 lines for now.

**Comment by LangLangBart** [comment, marked as answer]
> what is the escape sequance insside the `--header ' '` string that creates a new line?

Try it with `$'`[^1][^2]
```bash
--header $'foo\nbar'

```

> I also added --header-lines 2 if that does anything related I Just need 2 lines for now.

You shouldn't need `--header-lines`. It's more commonly used when your data may include a table
header and you don't want it to be selectable. For example:

```bash
ps aux | fzf --header 'Current processes' --header-lines 1
```

[^1]: [Bash Reference Manual - ANSI-C Quoting](https://www.gnu.org/software/bash/manual/bash.html#ANSI_002dC-Quoting)
[^2]: [bash quotes (wizardzines.com)](https://wizardzines.com/comics/bash-quotes/)

**Reply by MikeLemo1** [reply]
Huh ok yeah you needed `$` before the main string defenition `''` like `$'\n'` in order for escape and other special character and symbol tokenazation to occur / interpert the string in standard bash interpertation of the string symbols and tokens I did came across it but like a monkey tryed to use it inside the main header string like so:
```bash
--header 'CTRL-Y to yank $'/n' CTRL-T some other command'
``` 
and it  did interpolate it and removed the `$'\'` and left me with just the `n` in the end like so `CTRL-Y to yank n CTRL-T some other command`
but no new line :)

Thank you very much for all the help and dedicatred answers,
realy needed these tweaks and good explanation to fix my head messed with it for wayyy longer than it deserved.

**Reply by ernstki** [reply]
> and it did interpolate it and removed the `$'\'` and left me with just the `n` in the end like so `CTRL-Y to yank n CTRL-T some other command`

First thing to be aware of is this is a feature of Bash and other shells called [ANSI-C quoting](https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html) which [may not be supported elsewhere](https://unix.stackexchange.com/questions/371827/which-shells-support-ansi-c-quoting-e-g-string). (In Z shell it works, for the record.) As a shell feature, it has nothing to do with fzf itself.

Second, the `$'string with ANSI escape\nsequences\nin it'` _must_ be single-quoted, and the `$'` must begin the **outermost** set of quotes.

Taking all that into account, probably what you meant to do was:

```bash
fzf --header $'Header line one\nHeader line two'
```

![Screenshot demonstrating a two-line header using ANSI-C quoting](https://github.com/user-attachments/assets/517be495-042a-49c5-8573-1b5e90827cf5)

With a small amount of tinkering on your part, I'm confident you could've figured this out on your own. `:)`
